### PR TITLE
Updated "What's new" dialog

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: #6365 

Needs correct links and images.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/SplashDialog.cpp
Comment (vsverchinsky) [reply src/SplashDialog.cpp:1]: When compiling out of sources you need to explicitly define `-DSHOW_WHATS_NEW_SECTION=Yes` cmake variable to enable this new dialog. CI is configured to define that variable for us, but it's not the case when compiling out of sources. Linux distros maintainers likely compile Audacity with old-style `SplashDialog` without images and links to YouTube.

Comment (LWinterberg) [reply src/SplashDialog.cpp:1]: Right. Let's set SHOW_WHATS_NEW_SECTION=Yes in the source by default then, and let maintainers explicitly opt out of it if they don't want it. And if they do opt out, let's just not show any startup screen.

File: src/WhatsNewDialog.cpp
Comment (LWinterberg) [inline src/WhatsNewDialog.cpp:40]: MusehubURL and PluginsURL are identical. rename instances of PluginsURL to MuseHubURL and set MuseHubURL "https://www.musehub.com"

File: src/WhatsNewDialog.cpp
Comment (LWinterberg) [inline src/WhatsNewDialog.cpp:113]: this is a little mighty I found. Unless there's rendering differences between platforms, I found that a window height of 480 worked well with a 20px border.
 
before: 
image

after: 
image

File: src/WhatsNewDialog.cpp
Comment (LWinterberg) [inline src/WhatsNewDialog.cpp:140]: (this border I set to 20)

**Review by LWinterberg — APPROVED** [review]
still awaiting final approval for the strings from martin

**Comment by LWinterberg** [comment]
Oh also, while we wait for the strings: similar to #6446 we need to hide the musehub part of this on Linux because it's not available there. 

Might look something like this:

image

File: src/WhatsNewDialog.cpp
Comment (vsverchinsky) [reply src/WhatsNewDialog.cpp:113]: This is how it looks on my machine with 40 px border :/
изображение

Comment (vsverchinsky) [reply src/WhatsNewDialog.cpp:113]: Ah, this is because of dpi scaling ...

**Comment by LWinterberg** [comment]
Strings approved

**Comment by vsverchinsky** [comment]
@LWinterberg can you please check how it looks now for you?

**Comment by LWinterberg** [comment]
windows @ 100%
image

@ 150%

image

@ 225%
image

**Review by LWinterberg — APPROVED** [review]

**Comment by vsverchinsky** [comment]
Squashed and rebased

**Pull request merged**
